src/spacel/agent/systemd.py

```diff
@@ -34,13 +39,36 @@ def start_units(self, manifest):
                 continue
 
             try:
-                logger.debug('Starting "%s".', unit_id)
+                logger.info('Starting "%s".', unit_id)
                 unit.start('replace')
-                logger.debug('Started "%s".', unit_id)
+                waiting_units[unit_id] = time.time()
             except:
                 logger.warn('Error starting "%s".', unit_id, exc_info=True)
                 return False
 
+        while waiting_units:
+            for unit_name, unit_start in list(waiting_units.items()):
+                unit = self._manager.get_unit(unit_name)
+                unit_props = unit.properties
+
+                # Dump info to debug:
+                logger.debug('Service "%s" is: %s/%s', unit_name,
+                             unit_props.ActiveState, unit_props.SubState)
+
+                # Remove if completed:
+                if unit_props.ActiveState == 'active':
+                    duration = round(time.time() - unit_start, 2)
+                    logger.info('Started "%s" in %ss.', unit_name, duration)
+                    del waiting_units[unit_name]
```

Comment:
Implementation note: if `unit_props.SubState=='auto-restart'`, the service may be flapping (and will eventually timeout).

I considered making that state another early exit (thou shalt not flap!), but I wasn't sure if it was required for handling service dependencies.